src/index.ts

```diff
@@ -64,29 +64,31 @@ export function _buildMetadataAccessor(type: string) {
     rax.attach(ax);
     const baseOpts = {
       url: `${BASE_URL}/${type}${property}`,
-      headers: {'Metadata-Flavor': 'Google'},
+      headers: {'metadata-flavor': 'Google'},
       raxConfig: {noResponseRetries: 0}
     };
     const reqOpts = extend(true, baseOpts, options);
     delete (reqOpts as {property: string}).property;
-
     ax.request(reqOpts)
         .then(res => {
-          callback!(null, res, res.data);
+          if (res.headers['metadata-flavor'] !== 'Google') {
+            callback!(new Error(
+                `The 'metadata-flavor' header is not set to 'Google'.`));
```

Comment:
The error message is not useful for users. I would suggest something like: `Invalid response from metadata service: incorrect Metadata-Flavor header.`